lib/linter/node-event-generator.js

```diff
@@ -221,17 +221,23 @@ class NodeEventGenerator {
             const selector = parseSelector(rawSelector);
 
             if (selector.listenerTypes) {
-                selector.listenerTypes.forEach(nodeType => {
-                    const typeMap = selector.isExit ? this.exitSelectorsByNodeType : this.enterSelectorsByNodeType;
+                const typeMap = selector.isExit ? this.exitSelectorsByNodeType : this.enterSelectorsByNodeType;
 
+                selector.listenerTypes.forEach(nodeType => {
                     if (!typeMap.has(nodeType)) {
                         typeMap.set(nodeType, []);
                     }
                     typeMap.get(nodeType).push(selector);
                 });
-            } else {
-                (selector.isExit ? this.anyTypeExitSelectors : this.anyTypeEnterSelectors).push(selector);
+                return;
             }
+
+            if (selector.isExit) {
+                this.anyTypeExitSelectors.push(selector);
+                return;
+            }
+
+            this.anyTypeEnterSelectors.push(selector);
```

Comment:
I'm not sure the changed code has a better readability, I would rather something like:

```js
const selectors = selector.isExit ? this.anyTypeExitSelectors : this.anyTypeEnterSelectors;
selectors.push(selector);
```

Reply:
as you wish :) I updated it